src/platform/site-wide/side-nav/components/SideNav.js

```diff
@@ -0,0 +1,128 @@
+// Dependencies
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { find, filter, get, startsWith, map, orderBy } from 'lodash';
+
+class SideNav extends Component {
+  static propTypes = {
+    navItemsLookup: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      navItemsLookup: props.navItemsLookup,
+    };
+  }
+
+  toggleItemExpanded = id => () => {
+    const { navItemsLookup } = this.state;
+
+    // Derive if the item is expanded.
+    const expanded = get(navItemsLookup, `[${id}].expanded`);
+
+    // Flip the item's expanded property.
+    this.setState({
+      navItemsLookup: {
+        ...navItemsLookup,
+        [id]: {
+          ...navItemsLookup[id],
+          expanded: !expanded,
+        },
+      },
+    });
+  };
+
+  renderChildItems = (parentID, depth) => {
+    const { toggleItemExpanded } = this;
+    const { navItemsLookup } = this.state;
+
+    // Derive the items to render.
+    const filteredNavItems = filter(
+      navItemsLookup,
+      item => item.parentID === parentID,
+    );
+
+    // Sort the items by `order`.
+    const sortedNavItems = orderBy(filteredNavItems, 'order', 'asc');
+
+    return map(sortedNavItems, (item, index) => {
+      // Derive the item properties.
+      const expanded = get(item, 'expanded');
+      const hasChildren = get(item, 'hasChildren', false);
+      const href = get(item, 'href');
+      const id = get(item, 'id');
+      const label = get(item, 'label', '');
+
+      // Derive if the nav item is selected and the selected class.
+      const isSelected = startsWith(window.location.pathname, href);
+      const selectedClass = isSelected ? 'selected' : '';
+
+      // Caclculate the indentation for the child items.
+      const indentation = depth >= 2 ? 20 * (depth - 1) : 20;
+
+      // Determine if we should show a line at the end.
+      const showLine = depth === 1 && index !== sortedNavItems.length - 1;
+
+      return (
+        <li className={`va-sidenav-level-${depth}`} key={id}>
```

Comment:
Once we feel comfortable with the functionality this code produces, I can abstract these elements into their own functional components so it's more legible.